File: _src/03/Chapter3.CSharp/Person.cs


using System;
using System.Collections.Generic;

namespace Chapter3.CSharp
{
    public class Person
    {
public Person()
{
    Phones = new HashSet<Phone>();
}
        public int PersonId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string MiddleName { get; set; }
        public DateTime? BirthDate { get; set; }
        public decimal HeightInFeet { get; set; }
        public byte[] Photo { get; set; }
        public bool IsActive { get; set; }
        public int NumberOfCars { get; set; }
        public virtual ICollection<Phone> Phones { get; set; }
    }

    //public class Person
    //{
    //    public int PersonId { get; set; }

    //    [MaxLength(30, ErrorMessage = "First name cannot be longer than 30")]
    //    public string FirstName { get; set; }

    //    [MaxLength(30)]
    //    public string LastName { get; set; }

    //    [StringLength(1, MinimumLength = 1)]
    //    [Column(TypeName = "char")]
    //    public string MiddleName { get; set; }
    //}
}

File: _src/03/Chapter3.VB/Person.vb
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Public Class Person
Public Sub New
    Phones = new HashSet(Of Phone)
End Sub
    
    Property PersonId() As Integer
    Property FirstName() As String
    Property LastName() As String
    Property MiddleName() As String
    Property BirthDate() As DateTime?
    Property HeightInFeet() As Decimal
    Property Photo() As Byte()
    Property IsActive() As Boolean
    Property NumberOfCars() As Integer
    Overridable Property Phones() As ICollection(Of Phone)
End Class

'Public Class Person
'    Property PersonId() As Integer

'    <MaxLength(30, ErrorMessage:="First name cannot be longer than 30")>
'    Property FirstName() As String

'    <MaxLength(30)>
'    Property LastName() As String

'    <StringLength(1, MinimumLength:=1)>
'    <Column(TypeName:="char")>
'    Property MiddleName() As String

'End Class
